### DISCUSSION TITLE
Option to specify vercel config file path in web UI

### DISCUSSION THREAD

**Discussion by JeffersonBledsoe** [vercel/vercel#4359] [General] [answered]
Is there a currently a way to specify a path to the `vercel.json` or `now.json` file through the web UI? If not, any chance this could be added at some point?

My user case:
I'm experimenting with a toy project where I'd like to open-source the Next.JS app, however I need to keep the `vercel.json` file private. I realise I can use git submodules and a separate private GitHub repo to achieve this, however I would then need to resort to having CI deploy using the vercel CLI rather than the great GitHub integration which already exists within Vercel.

**Comment by styfle** [comment, marked as answer]
Yes, you can specify the "Root Directory" for your project which should contain the `vercel.json` file.

https://vercel.com/blog/advanced-project-settings

However, there are currently some limitations. When selecting the "Root Directory", all code above that directory is excluded. This means if you set `myapp` as the root directory with `/myapp/vercel.json` as the config, you won't be able to use `/example.js` or `/lib/helper.js`, only `/myapp/helper.js`.

We are working on better support for monorepos to support this use case.

**Comment by styfle** [comment]
That being said, why would `vercel.json` be private? It shouldn't contain any private information?

**Comment by JeffersonBledsoe** [comment]
Thanks for the help @styfle, I missed the Root Directory config option!

I was playing around with mimicking the [Netlify _redirects file URL shortener](https://dev.to/waylonwalker/personal-url-shortener-with-netlify-redirects-2co0) demo I've seen around but using the vercel.json file and vercel instead and wanted to see if I could make a public form whilst keeping the redirects that users have entered private.

**Comment by styfle** [comment]
Since you are using Next.js you can install the canary `npm i next@canary react react-dom` and utilize `next.config.js` for your redirects which has greater flexibility since its a js file that executes rather than a static json file.

https://github.com/zeit/next.js/issues/9081

**Comment by david1542** [comment]
Is there any updates on this issue?

I also use a monorepo and I have 2 applications hosted on vercel, located in apps/app-1 and apps/app-2. Each one has its own vercel.json and a separate app in Vercel. 

Now, I use the monorepo's root directory as the project root in Vercel since I need to build the monorepo and then build the apps, since the apps use some shared libraries.

Is there a way to specify just the config path in the dashboard?
